crates/jessie-parser/src/scope.rs

```diff
@@ -100,13 +85,29 @@ impl LexicalScope {
     fn visit_pattern_internal(&mut self, pattern: &Pattern, declaration_index: DeclarationIndex, property_access: &mut Vec<PropertyAccess>, is_hoisting_allowed: bool) -> Option<()> {
 
         match pattern {
-            Pattern::Rest(x) => unimplemented!("Rest pattern is not supported yet"),
+            Pattern::Rest(pattern) => {
+                // directly visit the inner pattern
+                self.visit_pattern_internal(pattern, declaration_index, property_access, is_hoisting_allowed)
+            }
             Pattern::Optional(opt) => {
-                unimplemented!("Optional pattern is not supported yet")
-                /* 
-                let OptionalPattern(_, jessie_ast::LValueOptional::Variable(var), default) = *opt;
-                self.declare(&name, declaration_index, property_access, Some(init), is_hoisting_allowed)
-                */
+                let OptionalPattern(_, LValueOptional::Variable(var), default) = opt;
```

Comment:
```suggestion
                // TODO: CPEAAPL approach has been changed from transmuting from lexer side analysis. 
                // We don't need to match the memory representations between pattern and expression, remove the first value later
                let OptionalPattern(_, LValueOptional::Variable(var), default) = opt;
```

Just memoing this for future me, ignore plz